File: Python/leetcode_1287.py
from collections import Counter
from typing import List


def find_special_integer(arr: List[int]) -> int:
    c = Counter(arr)
    return max(c, key=c.get)


print(find_special_integer([1, 2, 2, 6, 6, 6, 6, 7, 10]))

File: JavaScript/leetcode_1287.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var findSpecialInteger = function (arr) {
    let c = 0;
    let curr = arr[0];
    for (let index = 0; index < arr.length; index++) {
        if (arr[index] === curr) {
            c++;
            if (arr.length / 4 < c) return arr[index];
        } else {
            c = 1;
            curr = arr[index];
        }
    }
};
